src/Middleware/RequestBodyBuffer.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace React\Http\Middleware;
+
+use Psr\Http\Message\ServerRequestInterface;
+use React\Http\Response;
+use React\Promise\Stream;
+use React\Stream\ReadableStreamInterface;
+use RingCentral\Psr7\BufferStream;
+
+final class RequestBodyBuffer
+{
+    private $sizeLimit;
+
+    public function __construct($sizeLimit = null)
+    {
+        if ($sizeLimit === null) {
+            $sizeLimit = $this->iniMaxPostSize();
+        }
+
+        $this->sizeLimit = $sizeLimit;
+    }
+
+    public function __invoke(ServerRequestInterface $request, $stack)
+    {
+        $size = $request->getBody()->getSize();
+
+        if ($size === null) {
+            return new Response(411, array('Content-Type' => 'text/plain'), 'No Content-Length header given');
+        }
```

Comment:
Makes perfect sense to me, but we should probably add some documentation and tests for this :+1:

Reply:
Added a test for it

Reply:
And added docs for it